File: Iteration2/code/CSE3063GRP12/src/test/SessionControllerTest.java
package test;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import core.enums.UserType;
import core.general_providers.SessionController;
import core.models.concretes.Student;

public class SessionControllerTest {

    @Test
    public void testUserType() {
        SessionController sessionController = SessionController.getInstance();

        Student student = new Student();
        sessionController.setCurrentUser(student);

        UserType userType = sessionController.getUserType();

        assertEquals(userType, UserType.Student);

    }

    @Test
    public void testSetCurrentUser() {
        Student student = new Student();
        student.setUserName("test");
        SessionController sessionController = SessionController.getInstance();
        sessionController.setCurrentUser(student);
        assertEquals(sessionController.getCurrentUser(), student);
    }

}

File: Iteration3/code/CSE3063GRP12/src/SessionControllerTest.py
import unittest
from core.enums.UserType import UserType
from core.models.concretes.Student import Student
from core.general_providers.SessionController import SessionController
from core.models.concretes.Advisor import Advisor

class SessionControllerTest(unittest.TestCase):

    def test_user_type(self):
        session_controller = SessionController.getInstance()

        student = Student()
        advisor = Advisor()
        session_controller.set_current_user(student)

        user_type = session_controller.get_user_type()

        self.assertEqual(user_type, UserType.Student, "User type should be STUDENT")

    def test_set_current_user(self):
        student = Student()
        student.userName = "test"
        session_controller = SessionController.getInstance()
        session_controller.set_current_user(student)
        
        self.assertEqual(session_controller.get_current_user(), student, "Current user not set correctly")


if __name__ == '__main__':
    unittest.main()
    